src/Type/ContainerDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Type;
+
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Shamelessly copied from https://github.com/bnf/phpstan-psr-container/blob/master/src/ContainerDynamicReturnTypeExtension.php
+ * @author Benjamin Franzke
+ */
```

Comment:
Welcome to license hell... 
@bnf is it ok for you that we copy this piece of code here even if this package is MIT?

Reply:
While i was writing the comment, i knew it will be discussed in the PR.